### PR TITLE
TRT-1359: Add mechanism for creating flake junit for a monitor test

### PR DESCRIPTION
**Pull request by xueqzhan**

### CODE REVIEW

File: pkg/monitortestframework/impl.go [hunk 1]
```diff
@@ -115,7 +115,10 @@ func (r *monitorTestRegistry) StartCollection(ctx context.Context, adminRESTConf
 					},
 					SystemOut: fmt.Sprintf("failed during setup\n%v", err),
 				}
-				return
+				var flakeErr *NotSupportedError
+				if !errors.As(err, &flakeErr) {
+					return
+				}
```
Comment (stbenjam) [inline pkg/monitortestframework/impl.go:121]: Is *NotSupportedError the right type? I'm confused by that usage here

Comment (xueqzhan) [reply pkg/monitortestframework/impl.go]: Sigh.

**Pull request merged**